yt_dlp/extractor/bluesky.py

```diff
@@ -0,0 +1,342 @@
+            'alt_title': 'Bluesky video feature announcement',
+            'description': r're:Bluesky now has video!',
+            'upload_date': '20240911',
+            'timestamp': 1726074716,
+            'like_count': int,
+            'repost_count': int,
+            'comment_count': int,
+            'tags': [],
+            'language': 'en',
+            'subtitles': {
+                'en': 'mincount:1',
+            },
+        },
+    }, {
+        'url': 'https://bsky.app/profile/alt.bun.how/post/3l7rdfxhyds2f',
+        'md5': '8775118b235cf9fa6b5ad30f95cda75c',
+        'info_dict': {
+            'id': '3l7rdfxhyds2f',
+            'ext': 'mp4',
+            'uploader': 'cinnamon',
+            'uploader_id': 'alt.bun.how',
+            'uploader_url': 'https://bsky.app/profile/alt.bun.how',
+            'channel_id': 'did:plc:7x6rtuenkuvxq3zsvffp2ide',
+            'channel_url': 'https://bsky.app/profile/did:plc:7x6rtuenkuvxq3zsvffp2ide',
+            'thumbnail': r're:https://video.bsky.app/watch/.*\.jpg$',
+            'title': 'crazy that i look like this tbh',
+            'description': 'crazy that i look like this tbh',
+            'upload_date': '20241030',
+            'timestamp': 1730332128,
+            'like_count': int,
+            'repost_count': int,
+            'comment_count': int,
+            'tags': ['sexual'],
+            'age_limit': 18,
+            'language': 'en',
+        },
+    }, {
+        'url': 'at://did:plc:ia76kvnndjutgedggx2ibrem/app.bsky.feed.post/3l6zrz6zyl2dr',
+        'md5': '71b0eb6d85d03145e6af6642c7fc6d78',
+        'info_dict': {
+            'id': '3l6zrz6zyl2dr',
+            'ext': 'mp4',
+            'uploader': 'mary🐇',
+            'uploader_id': 'mary.my.id',
+            'uploader_url': 'https://bsky.app/profile/mary.my.id',
+            'channel_id': 'did:plc:ia76kvnndjutgedggx2ibrem',
+            'channel_url': 'https://bsky.app/profile/did:plc:ia76kvnndjutgedggx2ibrem',
+            'thumbnail': 'https://video.bsky.app/watch/did%3Aplc%3Aia76kvnndjutgedggx2ibrem/bafkreiasced5yaaodnspwgmuwvbxhoghbkq7iibhl3ftjgupq3brmgfwkm/thumbnail.jpg',
+            'title': 'Bluesky video #3l6zrz6zyl2dr',
+            'alt_title': '',
+            'description': '',
+            'upload_date': '20241021',
+            'timestamp': 1729523172,
+            'like_count': int,
+            'repost_count': int,
+            'comment_count': int,
+            'tags': [],
+            'language': 'en',
+        },
+    }]
+    _BLOB_URL = '%s/xrpc/com.atproto.sync.getBlob'
+
+    def get_service_endpoint(self, did, video_id):
+        if did.startswith('did:web:'):
+            url = f'https://{did.split(":")[-1]}/.well-known/did.json'
+        else:
+            url = f'https://plc.directory/{did}'
+        services = self._download_json(url, video_id, fatal=False)
+        return traverse_obj(
+            services, ('service', lambda _, x: x.get('type') == 'AtprotoPersonalDataServer',
```

Comment:
Can be fatal inside of traversals
```suggestion
            services, ('service', lambda _, x: x['type'] == 'AtprotoPersonalDataServer',
```